specs/packagemanager/PackageManagement.md

```diff
@@ -0,0 +1,1136 @@
+|Register     | WAS  |  OS/WAS  |       OS/WAS      |      OS/WAS     | OS/WAS |    X     | OS/WAS  |    WAS     |
+|Remove       |  X   |    X     |        WAS        |      OS/WAS     |   X    |    X     | OS/WAS  |    WAS     |
+|Repair       |  X   |    X     |        WAS        |       WAS       |   X    |    X     |  WAS    |    WAS     |
+|Reset        |  X   |    X     |        WAS        |       WAS       |   X    |    X     |  WAS    |    WAS     |
+|Provision    |  X   |    X     |       OS/WAS      |        X        |   X    |    X     |  WAS    |    WAS     |
+|Deprovision  |  X   |    X     |       OS/WAS      |        X        |   X    |    X     |  WAS    |    WAS     |
+
+Legend:
+
+* OS = Supported by Windows (OS) APIs in the Windows.Management.Deployment.PackageManager namespace.
+* WAS = Supported by Windows App SDK APIs in the
+  Microsoft.Windows.Management.Deployment.PackageDeploymentManager namespace.
+* X = Not supported
+
+## 3.2. Is*Ready()
+
+Is*Ready() methods determine if the target is installed and ready for use. Reasons why a package is
+not ready can include:
+
+* The package is not present on the machine
+* The package is present on the machine but not registered for the user
+* The package is registered for the user but is not in a healthy status e.g. its
+  Package.Status=Tampered
+
+Is*Ready() methods are a quick test to determine if more (costly) work is needed before the target
+can be used.
+
+## 3.3. Ensure*IsReady()
+
+Ensure*IsReady() methods determine if the target is installed and ready for use and, if not, makes
+it so. This can include downloading the target, registering it for the user and remediating a
+package in an unhealthy state.
+
+Thus `EnsurePackageIsReady(pkg, options)` is functionally equivalent to
+
+```c#
+var pdm = PackageDeploymentManager().GetDefault();
+if (!pdm.IsPackageReady(pkg))
+{
+    var result = await pdm.AddPackageAsync(pkg, options);
+}
+endif
+```
+
+### 3.3.1. Why Is*Ready() Given Ensure*IsReady()?
+
+Ensure*IsReady() performs an 'is ready' check and returns if all is ready. There's no efficiency
+reasons to call Is*Ready() before Ensure*IsReady() (in fact, it's less efficient as Is*Ready() would
+occur twice).
+
+However, this can be useful if you need additional work before potentially performing deployment
+operations. For example, if you need to prompt the user for consent before installing the target
+e.g.
+
+```c#
+var pdm = PackageDeploymentManager().GetDefault();
+if (!pdm.IsPackageReady(pkg))
+{
+    bool ok = AskUserForConsent(pkg);
+    if (ok)
+    {
+        var result = await pdm.EnsurePackageIsReadyAsync(pkg, options);
+    }
+}
+```
+
+## 3.4. Repair
+
+`PackageDeploymentManager` offers Repair APIs providing the same functionality as available
+interactively via Settings' `Repair` button on the detail page for an app (via Apps > Installed Apps
+> ... > Advanced options).
```

Comment:
Formatting issue: this line ends up as a blockquote

Reply:
More unintentional markdown fun. Thanks. Fixing